### PR TITLE
Improvements: Editor Validators

### PR DESCRIPTION
**Pull request by lciolecki**
- Improve BE error type
- Some FE improvements
- FE fix bug on create new process

### CODE REVIEW

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/context/ProcessCompilationError.scala [hunk 1]
```diff
@@ -134,13 +140,9 @@ object ProcessCompilationError {
       WrongParameters(requiredParameters, passedParameters, nodeId.id)
   }
 
-  case class ErrorValidationParameter(validator: ParameterValidator, paramName: String, nodeId: String)
-    extends PartSubGraphCompilationError with InASingleNode
+  case class BlankParameterError(message: String, description: String, paramName: String, nodeId: String) extends ParameterValidationError
```
Comment (PPiedel) [inline engine/api/src/main/scala/pl/touk/nussknacker/engine/api/context/ProcessCompilationError.scala]: It inherits from ParameterValidationError so BlankParameter / EmptyMandatoryParameter is sufficient, i think

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala [hunk 2]
```diff
@@ -629,7 +629,7 @@ class InterpreterSpec extends FunSuite with Matchers {
 
     intercept[IllegalArgumentException] {
       interpretSource(process, Transaction())
-    }.getMessage shouldBe "Compilation errors: ErrorValidationParameter(NotBlankParameterValidator,expression,customNode), " +
+    }.getMessage shouldBe "Compilation errors: BlankParameterError(Parameter expression value can't be blank,Please fill expression for this parameter,expression,customNode), " +
```
Comment (PPiedel) [inline engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala]: Again, we can use these validators for additional properties or fields, when "expression" does not exist. "value" should be sufficient? (also in EmptyMandatoryParameter, which I think should be renamed to EmptyMandatoryValue)

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/validation/PrettyValidationErrors.scala [hunk 3]
```diff
@@ -42,13 +37,8 @@ object PrettyValidationErrors {
...
         node(s"Node parameters not filled", s"Please fill missing node parameters: : ${params.mkString(", ")}")
 
-      // Parameter validation error cases
-      case ErrorValidationParameter(validator, paramName, _) => validator match {
-        case MandatoryParameterValidator =>
-          validation(validator, s"Parameter expression is mandatory", s"Please fill expression for this parameter", fieldName = Some(paramName))
-        case NotBlankParameterValidator =>
-          validation(validator, s"Parameter expression can't be blank", s"Please fill expression for this parameter", fieldName = Some(paramName))
-      }
+      // Parameter validation error case
```
Comment (PPiedel) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/validation/PrettyValidationErrors.scala]: do we need this comment?

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/context/ProcessCompilationError.scala (hunk 1 above)
Comment (lciolecki) [reply engine/api/src/main/scala/pl/touk/nussknacker/engine/api/context/ProcessCompilationError.scala]: Changed

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala (hunk 2 above)
Comment (lciolecki) [reply engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala]: You are right - I've forgotten about it..

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/validation/PrettyValidationErrors.scala (hunk 3 above)
Comment (lciolecki) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/validation/PrettyValidationErrors.scala]: Removed.

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala [hunk 4]
```diff
@@ -28,7 +28,14 @@ import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{ErrorVali
 case object MandatoryParameterValidator extends ParameterValidator {
 
   override def isValid(paramName: String, expression: String)(implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] =
-    if (StringUtils.isNotBlank(expression)) valid(Unit) else invalid(ErrorValidationParameter(this, paramName))
+    if (StringUtils.isNotBlank(expression)) valid(Unit) else invalid(error(paramName, nodeId.id))
+
+  private def error(paramName: String, nodeId: String): EmptyMandatoryParameter = EmptyMandatoryParameter(
+    "This field is mandatory and can not be empty",
+    "Please fill expression for this parameter",
```
Comment (PPiedel) [inline engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala]: value

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala [hunk 5]
```diff
@@ -39,7 +46,14 @@ case object NotBlankParameterValidator extends ParameterValidator {
 
   // TODO: for now we correctly detect only literal expression with blank string - on this level (not evaluated expression) it is the only thing that we can do
   override def isValid(paramName: String, expression: String)(implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] =
-    if (isBlankStringLiteral(expression)) invalid(ErrorValidationParameter(this, paramName)) else valid(Unit)
+    if (isBlankStringLiteral(expression)) invalid(error(paramName, nodeId.id)) else valid(Unit)
+
+  private def error(paramName: String, nodeId: String): BlankParameter = BlankParameter(
+    "This field value is required can not be blank",
+    "Please fill expression for this parameter",
```
Comment (PPiedel) [inline engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala]: value :P

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala [hunk 6]
```diff
@@ -617,7 +617,7 @@ class InterpreterSpec extends FunSuite with Matchers {
 
     intercept[IllegalArgumentException] {
       interpretSource(process, Transaction())
-    }.getMessage shouldBe "Compilation errors: ErrorValidationParameter(MandatoryParameterValidator,expression,customNode), " +
+    }.getMessage shouldBe "Compilation errors: EmptyMandatoryParameter(This field is mandatory and can not be empty,Please fill expression for this parameter,expression,customNode), " +
```
Comment (PPiedel) [inline engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala]: value

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala [hunk 7]
```diff
@@ -629,7 +629,7 @@ class InterpreterSpec extends FunSuite with Matchers {
 
     intercept[IllegalArgumentException] {
       interpretSource(process, Transaction())
-    }.getMessage shouldBe "Compilation errors: ErrorValidationParameter(NotBlankParameterValidator,expression,customNode), " +
+    }.getMessage shouldBe "Compilation errors: BlankParameter(This field value is required can not be blank,Please fill expression for this parameter,expression,customNode), " +
```
Comment (PPiedel) [inline engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala]: value

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala (hunk 5 above)
Comment (lciolecki) [reply engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala]: I will kill you ;)

**Pull request merged**